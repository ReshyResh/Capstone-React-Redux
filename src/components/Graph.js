import React from 'react';
import { Line } from 'react-chartjs-2';
import { useDispatch, useSelector } from 'react-redux';
import { fetchGraph } from '../redux/reducer/graphReducer';
import { useEffect } from 'react';
import Spinner from '../components/Spinner'

const LineChart = (props) => {
  const { coin } = props;
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchGraph(coin.symbol));
  }, [coin.symbol,dispatch]);
  const stats = useSelector((state) => state.graphReducer.stats);
  const loading = useSelector((state) => state.graphReducer.loading);
  const data = {
    labels: ["100m", "95m", "90m", "85m", "80m", "75m", "70m", "65m", "60m", "55m", "50m", "45m", "40m", "35m", "30m", "25m", "20m", "15m", "10m", "5m"],
    datasets: [
      {
        label: "Price",
        data: stats,
        fill: false,
        backgroundColor: "yellow",
        borderColor: "rgba(200, 200, 12, 0.5)",
      },
    ],
  };

  const options = {
    plugins: {
      legend: {
        labels: {
          color: "#fff",
          font: {
            size: 14
          }
        }
      }
    },
    scales: {
      y: { 
        ticks: {
          color: "#fff",
          font: {
            size: 14,
          },
          stepSize: 1,
          beginAtZero: true
        }
      },
      x: {
        ticks: {
          color: "#fff", 
          font: {
            size: 14
          },
          stepSize: 1,
          beginAtZero: true
        }
      }
    }
  }
  return (
    <>
    {loading ? <Spinner /> : null}
      <Line data={data} options={options} />
    </>
  );
};

export default LineChart;
