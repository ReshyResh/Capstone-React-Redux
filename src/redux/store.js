import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import coinReducer from './reducer/dataReducer';

const reducer = combineReducers({
  coinReducer,
});
const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);

export default store;