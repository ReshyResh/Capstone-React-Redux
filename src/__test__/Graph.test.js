import React from 'react';
import renderer from 'react-test-renderer';
import Graph from '../components/Graph';
import store from '../redux/store';
import { Provider } from 'react-redux';
import coin from './MockApiSingle';

it('Graph renders correctly', () => {
  const tree = renderer
    .create(<Provider store = { store }> <Graph coin = { coin } /> </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});