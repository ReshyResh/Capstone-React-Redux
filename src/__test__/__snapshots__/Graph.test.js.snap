// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph renders correctly 1`] = `
Array [
  " ",
  <div
    className="spinner-container"
  >
    <img
      alt="loading"
      className="spinner"
      src="loading.gif"
    />
  </div>,
  <canvas
    data-testid="canvas"
    height={150}
    onClick={[Function]}
    role="img"
    width={300}
  />,
  " ",
]
`;
